okhttp/src/main/kotlin/okhttp3/internal/cache/DiskLruCache.kt

```diff
@@ -397,10 +419,12 @@ class DiskLruCache(
     }
 
     if (fileSystem.exists(journalFile)) {
-      fileSystem.rename(journalFile, journalFileBackup)
+      fileSystem.atomicMove(journalFile, journalFileBackup)
+    }
+    fileSystem.atomicMove(journalFileTmp, journalFile)
+    if (fileSystem.exists(journalFileBackup)) {
```

Comment:
presumably this logic can be simplified by recognizing two cases:
 - journalFile exists on line 421 in which case the backup will exist here
 - journalFile doesn’t exist on line 421, in which case the backup shouldn’t exist here?

Reply:
Not convinced, but trusting your judgement.  Specifically it seems like "shouldn't exist here" is a weak assumption.